cmd/influxd/upgrade/config.go

```diff
@@ -146,20 +142,19 @@ func (cu *configUpgrader) updateV2Config(config map[string]interface{}, targetOp
 	}
 }
 
-func (cu *configUpgrader) save(config map[string]interface{}, path string) (string, error) {
-	buf := new(bytes.Buffer)
-	if err := toml.NewEncoder(buf).Encode(&config); err != nil {
-		return "", err
+func (cu *configUpgrader) save(config map[string]interface{}, path string) error {
+	// Open the target file, creating parent directories if needed.
+	if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
+		return err
 	}
-
-	err := ioutil.WriteFile(path, buf.Bytes(), 0666)
-	if err != nil { // permission issue possible - try to save the file to home or current dir
-		cu.log.Warn(fmt.Sprintf("Could not save upgraded config to %s, trying to save it to user's home.", path), zap.Error(err))
-		path = filepath.Join(homeOrAnyDir(), filepath.Base(path))
-		err = ioutil.WriteFile(path, buf.Bytes(), 0666)
+	outFile, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
is it okay to append to the file?

Reply:
The danger of copy-pasting code... the `upgrade` program asserts that there's no file at this target path before the logic reaches this point, but better to be consistent across the board. I'll update the flags to remove it.